### PR TITLE
Cfy 6734+6736 non root components

### PR DESCRIPTION
**Pull request by geokala**
Mgmtworker and manager IP setter

Includes some changes to improve IP setter.

### CODE REVIEW

File: components/manager-ip-setter/scripts/create.py [hunk 1]
```diff
@@ -15,30 +15,61 @@
...
 ctx_properties = utils.ctx_factory.create(MANAGER_IP_SETTER_SERVICE_NAME)
+MANAGER_IP_SETTER_USER = ctx_properties['os_user']
+MANAGER_IP_SETTER_GROUP = ctx_properties['os_group']
+HOMEDIR = ctx_properties['os_homedir']
+SUDOERS_INCLUDE_DIR = ctx_properties['sudoers_include_dir']
 
 MANAGER_IP_SETTER_SCRIPT_NAME = 'manager-ip-setter.sh'
 UPDATE_PROVIDER_CONTEXT_SCRIPT_NAME = 'update-provider-context.py'
-CREATE_INTERNAL_SSL_CERTS_SCRIPT_NAME = 'create-internal-ssl-certs.py'
+CREATE_SSL_CERTS_SCRIPT_NAME = 'create-ssl-certs.py'
```
Comment (idanmo) [inline components/manager-ip-setter/scripts/create.py]: Why did you rename this?

File: components/manager-ip-setter/scripts/manager-ip-setter.sh [hunk 2]
```diff
@@ -19,17 +19,27 @@ function set_manager_ip() {
...
 
   echo "Creating internal SSL certificates.."
-  /opt/cfy/embedded/bin/python /opt/cloudify/manager-ip-setter/create-internal-ssl-certs.py ${ip}
+  /opt/mgmtworker/env/bin/python /opt/cloudify/manager-ip-setter/create-ssl-certs.py ${ip}
   
-  echo "Restarting nginx.."
+  echo "Restarting services.."
+  # Restarting all (except postgres) to avoid issues with not correctly reloading SSL certs
   systemctl restart nginx
+  systemctl restart cloudify-rabbitmq
```
Comment (idanmo) [inline components/manager-ip-setter/scripts/manager-ip-setter.sh]: Why is it needed to restart all these services?
RabbitMQ already appears in line 45..

File: components/rabbitmq/config/rabbitmq-env.conf [hunk 3]
```diff
@@ -0,0 +1 @@
+NODENAME=cloudify-manager@localhost
```
Comment (idanmo) [inline components/rabbitmq/config/rabbitmq-env.conf]: Why is this needed?

File: components/rabbitmq/scripts/create.py [hunk 4]
```diff
@@ -81,6 +81,18 @@ def _install_rabbitmq():
...
+    # a manager is deployed from an image but given a new hostname.
+    # This is likely to cause problems with clustering of rabbitmq if this is
+    # done at any point, so at that point a change to the file and cleaning of
+    # mnesia would likely be necessary.
+    utils.deploy_blueprint_resource(
+        '{0}/rabbitmq-env.conf'.format(CONFIG_PATH),
+        '/etc/rabbitmq/rabbitmq-env.conf',
+        RABBITMQ_SERVICE_NAME)
+    # Delete old mnesia node
+    utils.sudo(['rm', '-rf', '/var/lib/rabbitmq/mnesia'])
```
Comment (idanmo) [inline components/rabbitmq/scripts/create.py]: why would this exist?

File: types/manager-types.yaml [hunk 5]
```diff
@@ -807,6 +819,22 @@ node_types:
...
+        default: cfyipsetter
+      os_group:
+        description: IP setter's OS group. Will be created if not existing, and used to run the service.
+        type: string
+        default: cfyipsetter
+      os_homedir:
+        description: IP setter's OS user's home directory. Will not be created if missing.
+        type: string
+        default: /etc/cloudify
+      sudoers_include_dir:
```
Comment (idanmo) [inline types/manager-types.yaml]: In general, please remember that if more users should be added to sudoers file, we should probably think of a better approach.. maybe as part of some setup node or some other idea.

File: components/manager-ip-setter/scripts/create.py (hunk 1 above)
Comment (geokala) [inline components/manager-ip-setter/scripts/create.py]: Because it is now regenerating the non-internal cert too.

File: components/manager-ip-setter/scripts/manager-ip-setter.sh (hunk 2 above)
Comment (geokala) [inline components/manager-ip-setter/scripts/manager-ip-setter.sh]: Whoops, I thought I removed the restarts from elsewhere.

I ran into issues on one of the build attempts where I had to restart some services that weren't restarted because at least one appeared to be caching a cert, so I put in restarts on all of the services we use so that even if we add extra logic to the IP setting at a later date, things should just work (tm), rather than having to decide specifically which services need restarting each time.

I'll remove the earlier RabbitMQ and retest.

File: components/rabbitmq/config/rabbitmq-env.conf (hunk 3 above)
Comment (geokala) [inline components/rabbitmq/config/rabbitmq-env.conf]: Without it, deploying the manager image with a different hostname to the one it had originally was causing rabbit to fail to start because it couldn't find the node (default is something like rabbit@<hostname>, and <hostname> was different.)

File: components/rabbitmq/scripts/create.py (hunk 4 above)
Comment (geokala) [inline components/rabbitmq/scripts/create.py]: Because rabbit has already been started at this point, and not removing it will leave the old named mnesia datastore.

File: types/manager-types.yaml (hunk 5 above)
Comment (geokala) [inline types/manager-types.yaml]: Potentially, yes. I'm not sure how many we're going to need to add on this run though, so adding extra complexity and retrieving the sudoers include dir path from a relationship is probably overkill until we add another one?
